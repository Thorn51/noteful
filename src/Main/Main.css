.note {
  margin: 10px;
  border: lightgray solid 1px;
}

.note_name {
  font-size: 1.5rem;
  margin: 10px;
}

.note_modification {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.mod_date {
  margin: 10px;
}

.delete_note {
  margin: 10px;
}
