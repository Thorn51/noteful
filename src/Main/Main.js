import React from "react";
import "./Main.css";
import store from "../store";
import moment from "moment";

export default class Main extends React.Component {
  render() {
    const folderList = Object.keys(store.folders).map(key => {
      return (
        <div key={store.folders[key].id} className="folder">
          <h2>{store.folders[key].name}</h2>
        </div>
      );
    });

    const noteList = Object.keys(store.notes).map(key => {
      return (
        <div key={store.notes[key].id} className="note">
          <h2>{store.notes[key].name}</h2>
          <p>
            Date modified:{" "}
            {moment(store.notes[key].modified).format("Do, MMM YYYY")}
          </p>
          <button>Delete</button>
        </div>
      );
    });
    return (
      <div className="main_container">
        <section className="main_sidebar">
          <h2>Folder Section</h2>
          {folderList}
          <button className="main_add_folder_button">Add Folder</button>
        </section>
        <section className="main_content">
          <h2>Notes Section</h2>
          {noteList}
          <button className="main_add_note_button">Add Note</button>
        </section>
      </div>
    );
  }
}
