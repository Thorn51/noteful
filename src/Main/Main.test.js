import React from "react";
import ReactDOM from "react-dom";
import Main from "./Main";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

describe("Main", () => {
  it("renders without throwing errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Main />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the component as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Main />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
