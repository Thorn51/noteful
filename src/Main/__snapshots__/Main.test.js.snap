// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders the component as expected 1`] = `
<section
  className="notes_section"
>
  <div
    className="note_list"
  />
  <a
    href="/add-note"
    onClick={[Function]}
  >
    <button
      className="add_note"
    >
      Add Note
    </button>
  </a>
</section>
`;
